x-pack/plugins/alerting/server/plugin.ts

```diff
@@ -286,18 +285,21 @@ export class AlertingPlugin {
           ActionGroupIds,
           RecoveryActionGroupId
         >
-      ) {
+      ) => {
         if (!(ruleType.minimumLicenseRequired in LICENSE_TYPE)) {
           throw new Error(`"${ruleType.minimumLicenseRequired}" is not a valid license type`);
         }
         ruleType.config = getExecutionConfigForRuleType({
-          config: alertingConfig.rules,
+          config: this.config.rules,
+          ruleTypeId: ruleType.id,
+        });
+        ruleType.ruleTaskTimeout = getRuleTaskTimeout({
+          config: this.config.rules,
+          ruleTaskTimeout: ruleType.ruleTaskTimeout,
           ruleTypeId: ruleType.id,
         });
-        ruleType.ruleTaskTimeout =
-          ruleType.ruleTaskTimeout ?? alertingConfig.defaultRuleTaskTimeout;
         ruleType.cancelAlertsOnRuleTimeout =
-          ruleType.cancelAlertsOnRuleTimeout ?? alertingConfig.cancelAlertsOnRuleTimeout;
+          ruleType.cancelAlertsOnRuleTimeout ?? this.config.cancelAlertsOnRuleTimeout;
         ruleType.doesSetRecoveryContext = ruleType.doesSetRecoveryContext ?? false;
         ruleTypeRegistry.register(ruleType);
```

Comment:
Not necessary for this PR but might be nice eventually to refactor this logic into a single function and call it as the first step in `ruleTypeRegistry.register()`.

Reply:
will create a sub-issue under refactoring meta-issue for this.